tests/ui/unnecessary_cast_fixable.stderr

```diff
@@ -18,5 +18,35 @@ error: casting integer literal to `f64` is unnecessary
 LL |     100_i32 as f64;
    |     ^^^^^^^^^^^^^^ help: try: `100_f64`
 
-error: aborting due to 3 previous errors
+error: casting integer literal to `u32` is unnecessary
+  --> $DIR/unnecessary_cast_fixable.rs:22:5
+   |
+LL |     1 as u32;
+   |     ^^^^^^^^ help: try: `1_u32`
+
+error: casting integer literal to `i32` is unnecessary
+  --> $DIR/unnecessary_cast_fixable.rs:23:5
+   |
+LL |     0x10 as i32;
+   |     ^^^^^^^^^^^ help: try: `16_i32`
```

Comment:
For integers, we should not use decimal in the suggestion if the literal was using hexadecimal, octal or binary.

For floats we should make sure we only lint literals using decimal to mimick the behavior of the previous implementation of the lint (see #5220)

Reply:
About the floats, isn't it already the case since we're not linting casts of int literals to float types ? (If my understanding is correct, the idea is to avoid adding a type suffix that happens to be valid hex to an hex literal).

Reply:
>About the floats, isn't it already the case

Oh my bad, I was thinking about the edge cases of the previous implementation (e.g. no leading zeros for the from type), but as you say those are not covered in the enhancement.

Reply:
Done